s3parq/publish_parq.py

```diff
@@ -503,3 +538,105 @@ def publish(bucket: str, key: str, partitions: List[str], dataframe: pd.DataFram
     logger.info("Done writing to S3.")
 
     return files
+
+def custom_publish(bucket: str, key: str, partitions: List[str], dataframe: pd.DataFrame, custom_redshift_columns: dict, redshift_params: dict = None) -> List[str]:
+    """ Dataframe to S3 Parquet Publisher with a CUSTOM redshift column definition.
+    This function handles the portion of work that will see a dataframe converted
+    to parquet and then published to the given S3 location.
+    It supports partitions and will use the custom redshift columns defined in the
+    custom_redshift_columns dictionary when creating the table schema for the parquet file.
+    It also has the option to automatically publish up to Redshift Spectrum for
+    the newly published parquet files.
+
+    Args:
+        bucket (str): S3 Bucket name
+        key (str): S3 key to lead to the desired dataset
+        partitions (List[str]): List of columns that should be partitioned on
+        dataframe (pd.DataFrame): Dataframe to be published
+        custom_redshift_columns (dict): 
+            This dictionary contains custom column data type definitions for redshift.
+            The params should be formatted as follows:
+                - column name (str)
+                - data type (str)
```

Comment:
On top of the Decimal support comment above, can we list where to find which data types can be used? Like where to tell if I should pass int, integer, INT, etc

Reply:
Thanks! Addressed this in the readme and pointed there.